infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -550,10 +596,41 @@ public static LogUnitServerConfig parse(Map<String, Object> opts) {
             return LogUnitServerConfig.builder()
                     .cacheSizeHeapRatio(cacheSizeHeapRatio)
                     .maxCacheSize((long) (Runtime.getRuntime().maxMemory() * cacheSizeHeapRatio))
-                    .memoryMode(Boolean.valueOf(opts.get("--memory").toString()))
+                    .memoryMode(Boolean.parseBoolean(opts.get("--memory").toString()))
                     .noVerify((Boolean) opts.get("--no-verify"))
                     .noSync((Boolean) opts.get("--no-sync"))
                     .build();
         }
     }
+
+    /**
+     * Utility class used by the LogUnit server to initialize its components,
+     * including the StreamLog, LogUnitServerCache, BatchProcessor and StreamLogCompaction.
+     * This facilitates the injection of mocked objects during unit tests.
+     */
+    public static class LogUnitServerInitializer {
+        StreamLog buildStreamLog() {
```

Comment:
the naming can be improved here, consider renaming it to `buildInMemoryStreamLog`